### ISSUE TITLE
`parentBackground` breaks last symbol in `powerline` and `diamond`

### ISSUE THREAD

**Issue by thanhph111** [JanDeDobbeleer/oh-my-posh#1433]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

1. Let's take the default _Aliens_ theme.
2. Change a `background` field to `parentBackground` like in my images below.
3. We've got a glitch.

| Screenshot with git | Screenshot without git |
| --- | --- |
| ![Screenshot with git](https://user-images.githubusercontent.com/60842560/146043182-0b72973c-f4a5-4f19-af0c-5bd7166827ed.png) | ![Screenshot without git](https://user-images.githubusercontent.com/60842560/146043272-12bb4f1c-bee0-4a8b-9ce2-d0565eb293c4.png) |

I've been using _Oh My Posh_ from the very beginning, just upgraded from an old version. While it doesn't give me the same depth of control as the _PowerShell_ version did, to be honest, it still makes me very happy. Thank you for creating a great tool.

### Theme

```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "type": "session",
          "style": "diamond",
          "foreground": "#ffffff",
          "background": "#61AFEF",
          "leading_diamond": "\uE0B6",
          "trailing_diamond": "\uE0B0"
        },
        {
          "type": "path",
          "style": "powerline",
          "powerline_symbol": "\uE0B0",
          "foreground": "#ffffff",
          "background": "parentBackground",
          "properties": {
            "style": "full"
          }
        },
        {
          "type": "git",
          "style": "powerline",
          "powerline_symbol": "\uE0B0",
          "foreground": "#193549",
          "background": "#95ffa4",
          "properties": {
            "template": "{{ .HEAD }}"
          }
        },
        {
          "type": "python",
          "style": "diamond",
          "foreground": "#ffffff",
          "background": "parentBackground",
          "leading_diamond": "<transparent,background>\uE0B0</>",
          "trailing_diamond": "\uE0B4"
        }
      ]
    }
  ],
  "final_space": true
}
```

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash, powershell

### Log output

```shell
Version: 6.33.0

Segments:

ConsoleTitle(false) -   0 ms - 
session(true)  -   0 ms -  thanhph111@T-1000 
path(true)     -   0 ms -  ~/Desktop/temp 
git(true)      -   0 ms -  master 
python(false)  -   0 ms - 

Run duration: 693.702µs

Logs:

2021/12/14 23:50:44 debug: getenv

2021/12/14 23:50:44 getenv duration: 4.615µs, args: XDG_CACHE_HOME
2021/12/14 23:50:44 getCachePath duration: 16.294µs, args: 
2021/12/14 23:50:44 getArgs duration: 57ns, args: 
2021/12/14 23:50:44 getArgs duration: 29ns, args: 
2021/12/14 23:50:44 getShellName duration: 190ns, args: 
2021/12/14 23:50:44 debug: getenv

2021/12/14 23:50:44 getenv duration: 1.164µs, args: OMP_CACHE_DISABLED
2021/12/14 23:50:44 isRunningAsRoot duration: 543ns, args: 
2021/12/14 23:50:44 getcwd duration: 14.346µs, args: 
2021/12/14 23:50:44 getPathSeperator duration: 46ns, args: 
2021/12/14 23:50:44 getPathSeperator duration: 27ns, args: 
2021/12/14 23:50:44 getPathSeperator duration: 28ns, args: 
2021/12/14 23:50:44 getPathSeperator duration: 30ns, args: 
2021/12/14 23:50:44 getPathSeperator duration: 42ns, args: 
2021/12/14 23:50:44 getPathSeperator duration: 29ns, args: 
2021/12/14 23:50:44 getShellName duration: 66ns, args: 
2021/12/14 23:50:44 getCurrentUser duration: 154ns, args: 
2021/12/14 23:50:44 getHostName duration: 1.31µs, args: 
2021/12/14 23:50:44 getcwd duration: 82ns, args: 
2021/12/14 23:50:44 debug: getenv

2021/12/14 23:50:44 getenv duration: 989ns, args: SSH_CONNECTION
2021/12/14 23:50:44 debug: getenv

2021/12/14 23:50:44 getenv duration: 770ns, args: SSH_CLIENT
2021/12/14 23:50:44 getCurrentUser duration: 188ns, args: 
2021/12/14 23:50:44 getRuntimeGOOS duration: 46ns, args: 
2021/12/14 23:50:44 getHostName duration: 873ns, args: 
2021/12/14 23:50:44 debug: getenv

2021/12/14 23:50:44 getenv duration: 606ns, args: POSH_SESSION_DEFAULT_USER
2021/12/14 23:50:44 getcwd duration: 58ns, args: 
2021/12/14 23:50:44 getcwd duration: 27ns, args: 
2021/12/14 23:50:44 getArgs duration: 33ns, args: 
2021/12/14 23:50:44 getcwd duration: 31ns, args: 
2021/12/14 23:50:44 getRuntimeGOOS duration: 33ns, args: 
2021/12/14 23:50:44 getRuntimeGOOS duration: 29ns, args: 
2021/12/14 23:50:44 getPathSeperator duration: 49ns, args: 
2021/12/14 23:50:44 getRuntimeGOOS duration: 28ns, args: 
2021/12/14 23:50:44 getcwd duration: 37ns, args:
```

**Comment by JanDeDobbeleer** [comment]
@thanhph111 this syntax is actually not supported as we do not loop the colors:

```json
"background": "parentBackground",
"leading_diamond": "<transparent,background>\uE0B0</>",
```

it means in the diamond, `background` translates to `parentBackground` which isn't resolved to the actual color and falls back to the current bg (which is correct, so it ends up being a full coloured block).

Changing it to this fixes the issue:

```json
"background": "parentBackground",
"leading_diamond": "<transparent,parentBackground>\uE0B0</>",
```

I'll see what I can do ;-)

**Comment by thanhph111** [comment]
Sorry for picking a special example, let me take another one here:

```json
{
    "blocks": [
        {
            "alignment": "left",
            "newline": false,
            "type": "prompt",
            "segments": [
                {
                    "type": "time",
                    "style": "diamond",
                    "leading_diamond": "\uE0B6",
                    "background": "red",
                    "foreground": "lightWhite"
                },
                {
                    "type": "session",
                    "style": "powerline",
                    "powerline_symbol": "\uE0B4",
                    "foreground": "parentForeground",
                    "background": "yellow",
                    "properties": {
                        "template": "{{ .ComputerName }}"
                    }
                },
                {
                    "type": "shell",
                    "style": "powerline",
                    "foreground": "parentForeground",
                    "background": "parentBackground",
                    "properties": {
                        "prefix": "● "
                    }
                },
                {
                    "type": "path",
                    "style": "powerline",
                    "powerline_symbol": "\uE0B4",
                    "foreground": "lightWhite",
                    "background": "blue"
                }
            ]
        }
    ]
}
```

And we will have:

![image](https://user-images.githubusercontent.com/60842560/146309035-f5b86da2-eb34-49a0-a9aa-ca296cdc6984.png)

As you said, we don't loop to expand values so only first `parentForeground` works, that's correct. However, the `parentBackground` doesn't seem to expand either. It lets the last symbol fall back to default background color. I don't know if it was intended or not.

Changing back to specific color works perfectly:

```diff
-                    "background": "parentBackground",
+                    "background": "yellow",
```

![image](https://user-images.githubusercontent.com/60842560/146309748-ff1c3808-2b5e-440f-9fdb-4df319cf89e1.png)


```diff
-                    "background": "parentBackground",
+                    "background": "green",
```
![image](https://user-images.githubusercontent.com/60842560/146309779-139c5198-2b48-439f-a79e-ed264dfba2fe.png)

**Comment by JanDeDobbeleer** [comment]
@thanhph111 the last example has one thing which isn't supported and that's a grandparent. But I'm working on resolving the issue as we could in theory support it.

**Comment by thanhph111** [comment]
Thanks a lot. I'm not familiar with Go so I'm not really useful with this, but if you need anything more I'd be happy to help. And yeah, the "grandparent" feature should be in another issue.

**Comment by JanDeDobbeleer** [comment]
@thanhph111 I fixed it, but it needs proper testing. This part of the codebase isn't as easy, but I was able to make it simpler in the process.

**Comment by thanhph111** [comment]
Glad to hear that, thank you.
BTW, just curious, I've seen a high frequency of releases, how do you decide which version to choose for your next release?

**Comment by JanDeDobbeleer** [comment]
Every change gets pushed, versions are generated automatically.

**Comment by thanhph111** [comment]
So we aren't following semantic versioning?

**Comment by JanDeDobbeleer** [comment]
@thanhph111 yes we are, fully automated via conventional commits.

**Comment by thanhph111** [comment]
@JanDeDobbeleer Does version 6.36.1 include this issue? Looks like it's still broken somehow.

![image](https://user-images.githubusercontent.com/60842560/146482169-b7db2d48-d25a-4d68-9e10-dce5ea8c02fd.png)

**Comment by thanhph111** [comment]
@JanDeDobbeleer Does version 6.36.1 include this issue? Looks like it's still broken somehow.

![image](https://user-images.githubusercontent.com/60842560/146482620-e409f3ad-4dba-4311-85c9-31f16311433a.png)

**Comment by JanDeDobbeleer** [comment]
Yes it does: 

<img width="252" alt="image" src="https://user-images.githubusercontent.com/2492783/146498245-4293cd03-1bdd-4bf4-a5ad-682a2b25187d.png">

In your sample the shell segment doesn't have a `powerline_symbol` so we obviously do not print anything. That was actually a bug, now we do it correctly.

**Comment by thanhph111** [comment]
Sorry for disturbing you again. I try to reproduce this old settings on this new release style but no luck. Here my config:

```json
{
    "blocks": [
        {
            "alignment": "left",
            "newline": false,
            "type": "prompt",
            "segments": [
                {
                    "type": "time",
                    "style": "diamond",
                    "leading_diamond": "\ue0b6",
                    "background": "red",
                    "foreground": "lightWhite"
                },
                {
                    "type": "session",
                    "style": "powerline",
                    "powerline_symbol": "\ue0b4",
                    "foreground": "lightWhite",
                    "background": "yellow",
                    "properties": {
                        "template": "{{ .ComputerName }}"
                    }
                },
                {
                    "type": "shell",
                    "style": "powerline",
                    "foreground": "lightWhite",
                    "background": "yellow",
                    "properties": {
                        "prefix": "● "
                    }
                },
                {
                    "type": "envvar",
                    "style": "powerline",
                    "foreground": "lightWhite",
                    "background": "yellow",
                    "properties": {
                        "prefix": "",
                        "template": "\uf708",
                        "var_name": "NO_HISTORY"
                    }
                },
                {
                    "type": "path",
                    "style": "powerline",
                    "powerline_symbol": "\ue0b4",
                    "foreground": "lightWhite",
                    "background": "blue"
                }
            ]
        }
    ]
}
```

Here are the renders:

![image](https://user-images.githubusercontent.com/60842560/146505427-dc6ef5d6-8dec-4193-81fb-7115a2ace9f3.png)

![image](https://user-images.githubusercontent.com/60842560/146505441-c326ddf9-e1d5-45e6-9994-ee8f16505947.png)

I have tried several configurations but none of them are the same as my old ones, they have an extra space between segments.

![image](https://user-images.githubusercontent.com/60842560/146507364-9549b774-08c6-470d-9a4e-0d000c908dc2.png)

![image](https://user-images.githubusercontent.com/60842560/146507377-1b024788-e0a0-4ed4-848a-b2e2105d5704.png)

**Comment by thanhph111** [comment]
@JanDeDobbeleer I saw the release, thank you very much, but it doesn't seem to change, do I have to rewrite the config or something?

![image](https://user-images.githubusercontent.com/60842560/146557007-796147e5-fe7f-49e1-bdbc-4dfb8fb6ea78.png)

But it's quite not important, I can live with 6.34.1 for a while, if you feel tired with this issue, we can fix it later, no need to hurry.

**Comment by JanDeDobbeleer** [comment]
I don't feel tired, I did fix it but maybe I did something stupid.

**Comment by JanDeDobbeleer** [comment]
@thanhph111 I just double checked and it is fixed on my end with [6.36.2](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v6.36.2), using the [config](https://github.com/JanDeDobbeleer/oh-my-posh/issues/1433#issuecomment-996501850) you provided I get the following result:

<img width="321" alt="image" src="https://user-images.githubusercontent.com/2492783/146573844-f8ec5cab-7db4-43a6-8388-8c365cde6b45.png">

<img width="321" alt="image" src="https://user-images.githubusercontent.com/2492783/146573868-d557cbf5-970f-4130-a2d7-0be593fd5fba.png">

**Comment by thanhph111** [comment]
Sorry for that, I've just realized I don't have to put `powerline_symbol` to followed segments anymore, that's amazing. Thanks a lot, man. You're brilliant.
